span.go

```diff
@@ -129,6 +139,34 @@ func (s *spanImpl) FinishWithOptions(opts opentracing.FinishOptions) {
 
 	s.Lock()
 	defer s.Unlock()
+
+	if s.numDroppedLogs > 0 {
+		// We dropped some log events, which means that we used Logs as a circular
+		// buffer. De-circularize it.
+		pos := s.numDroppedLogs % s.tracer.options.MaxLogsPerSpan
+		// The oldest event is on position pos. We rotate (circular shift) the
+		// buffer in place, moving the first pos elements to the back. This
+		// algorithm is described in:
+		// 	http://www.cplusplus.com/reference/algorithm/rotate
+		for first, middle, next := 0, pos, pos; first != middle; {
+			s.raw.Logs[first], s.raw.Logs[next] = s.raw.Logs[next], s.raw.Logs[first]
+			first++
+			next++
+			if next == len(s.raw.Logs) {
+				next = middle
+			} else if first == middle {
+				middle = next
+			}
+		}
+		// Replace the oldest event. This means we are effectively dropping one more
+		// event.
+		s.raw.Logs[0].Event = fmt.Sprintf("** dropped %d events **", s.numDroppedLogs+1)
```

Comment:
FYI, you have unlucky timing... this is not going to compile at `master` since the OpenTracing KV logging thing just landed today.

Thankfully, this is a good case for KV logging :)